### DISCUSSION TITLE
How to use konva.js in a Mini Program ?

### DISCUSSION THREAD

**Discussion by Bertha-zxd** [konvajs/konva#1192] [Q&A]
DOM operations is forbidden by mini program in wechat, so neither createElement nor stage. add (layer) can be used, but we do can operate canvas and obtain the dom of canvas. How can we skip stage and use konva? It would be better if there were a better solution.

**Comment by lavrton** [comment]
I guess you will need to dive-in into that. I have no experience with WeChat. So, I don't know how to adopt for that. Konva is using `document.createElement('canvas')` a lot. It creates two canvases for each layer and may create a lot more if you have cached shapes.

**Reply by Bertha-zxd** [reply]
Thank you for your explanation.
